mobile_wallet/src/lib.rs

```diff
@@ -163,6 +194,166 @@ fn create_encrypted_transfer_aux(input: &str) -> anyhow::Result<String> {
+                .contracts
+                .get(contract_name)
+                .ok_or_else(|| anyhow::anyhow!("Unable to find contract inside module"))?;
+
+            contract_schema
+                .receive
+                .get(entrypoint_name)
+                .ok_or_else(|| anyhow::anyhow!("Unable to find receive schema"))?
+                .clone()
+        }
+        VersionedModuleSchema::V1(module_schema) => {
+            let contract_schema = module_schema
+                .contracts
+                .get(contract_name)
+                .ok_or_else(|| anyhow::anyhow!("Unable to find contract inside module"))?;
+
+            let entrypoint_parameter = contract_schema
+                .receive
+                .get(entrypoint_name)
+                .ok_or_else(|| anyhow::anyhow!("Unable to find receive schema"))?
+                .parameter();
+            match entrypoint_parameter {
+                Some(value) => value.clone(),
+                None => return Err(anyhow::anyhow!("Missing parameter for entrypoint")),
+            }
+        }
+        VersionedModuleSchema::V2(module_schema) => {
+            let contract_schema = module_schema
+                .contracts
+                .get(contract_name)
+                .ok_or_else(|| anyhow::anyhow!("Unable to find contract inside module"))?;
+
+            let entrypoint_parameter = contract_schema
+                .receive
+                .get(entrypoint_name)
+                .ok_or_else(|| anyhow::anyhow!("Unable to find receive schema"))?
+                .parameter();
+            match entrypoint_parameter {
+                Some(value) => value.clone(),
+                None => return Err(anyhow::anyhow!("Missing parameter for entrypoint")),
+            }
+        }
+    };
+    Ok(receive_schema)
+}
+
+fn get_parameters_as_json(
+    payload: &UpdateContractPayload,
+    schema: &str,
+    schema_version: &Option<u8>,
+) -> anyhow::Result<Value> {
+    let mut schema_bytes = hex::decode(schema)?;
+
+    let contract_name = &payload
+        .receive_name
+        .as_receive_name()
+        .contract_name()
+        .to_string();
+    let entrypoint_name = &payload
+        .receive_name
+        .as_receive_name()
+        .entrypoint_name()
+        .to_string();
+
+    let receive_schema = match from_bytes::<VersionedModuleSchema>(&schema_bytes) {
+        Ok(versioned) => get_receive_schema(versioned, contract_name, entrypoint_name)?,
+        Err(_) => match schema_version {
+            Some(version) => {
+                let mut versioned_schema_bytes: Vec<u8> = vec![u8::MAX, u8::MAX, *version];
+                versioned_schema_bytes.append(&mut schema_bytes);
+                let versioned_schema: VersionedModuleSchema =
+                    match from_bytes(&versioned_schema_bytes) {
+                        Ok(v) => v,
+                        Err(_) => return Err(anyhow::anyhow!("Invalid schema or schema version")),
+                    };
```

Comment:
Instead of this we could also do
```
match schema_version {
    0 => Ok(VersionedModuleSchema::V0(from_bytes(&schema_bytes)?),
    1 => Ok(...)
    2 => Ok(...)
}
```

Which at least looks a bit less hacky. If you don't think this is better, then please add documentation to the current approach of prepending these bits.